### ISSUE TITLE
VUMPS/TDVP broken for 1-site unit cell

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensorInfiniteMPS.jl#42]
When I run VUMPS/TDVP with a 1-site unit cell, it gives the following error:
```julia
ERROR: LoadError: ArgumentError: Collection is empty, must contain exactly 1 element
Stacktrace:
 [1] only(x::Vector{Index{Vector{Pair{QN, Int64}}}})
   @ Base.Iterators ./iterators.jl:1354
 [2] tdvp_iteration_parallel(solver::typeof(ITensorInfiniteMPS.vumps_solver), ∑h::InfiniteITensorSum, ψ::InfiniteCanonicalMPS; ϵᴸ!::Vector{Float64}, ϵᴿ!::Vector{Float64}, time_step::Float64, solver_tol::var"#136#137")
   @ ITensorInfiniteMPS ~/.julia/dev/ITensorInfiniteMPS/src/vumps_localham.jl:505
 [3] tdvp_iteration(::Function, ::Vararg{Any}; multisite_update_alg::String, kwargs::Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:ϵᴸ!, :ϵᴿ!, :solver_tol, :time_step), Tuple{Vector{Float64}, Vector{Float64}, var"#136#137", Float64}}})
   @ ITensorInfiniteMPS ~/.julia/dev/ITensorInfiniteMPS/src/vumps_localham.jl:295
 [4] tdvp(solver::Function, ∑h::InfiniteITensorSum, ψ::InfiniteCanonicalMPS; maxiter::Int64, tol::Float64, outputlevel::Int64, multisite_update_alg::String, time_step::Float64, solver_tol::Function)
   @ ITensorInfiniteMPS ~/.julia/dev/ITensorInfiniteMPS/src/vumps_localham.jl:647
 [5] tdvp(::InfiniteITensorSum, ::Vararg{Any}; time_step::Float64, solver_tol::Function, kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:tol, :maxiter, :multisite_update_alg), Tuple{Float64, Int64, String}}})
   @ ITensorInfiniteMPS ~/.julia/dev/ITensorInfiniteMPS/src/vumps_localham.jl:702
 [6] macro expansion
   @ ~/.julia/dev/ITensorInfiniteMPS/examples/vumps/vumps_ising.jl:57 [inlined]
 [7] top-level scope
   @ ./timing.jl:220
 [8] include(fname::String)
   @ Base.MainInclude ./client.jl:451
 [9] top-level scope
   @ REPL[7]:1
in expression starting at /home/mfishman/.julia/dev/ITensorInfiniteMPS/examples/vumps/vumps_ising.jl:53
```

@LHerviou it looks like it is erroring on some of the non-local interaction code introduced in #31.

**Comment by LHerviou** [comment]
I think I see where the problem lies.
In some of the deltas, there are a lot of only, which are not working well when the sites are all identical.
I can correct that quickly tomorrow morning.

**Comment by mtfishman** [comment]
Ok thanks! No rush. I'll add some placeholder tests for things like this (I'll add a loop over the unit cell size, but not include `1` for now).

**Comment by LHerviou** [comment]
I was wrong on the origin. It seems like something broke in the subspace expansion actually, specifically for N = 1 (if it was ever working, I do not remember exactly). It is pretty weird because I really did not change the part of the code that leads to the bug, as far as I remember

**Comment by mtfishman** [comment]
Ah interesting. If I remember correctly, the code was working for `N = 1` before these three recent PRs, but there were a lot of changes made.

**Comment by LHerviou** [comment]
So I just tried with all the versions of the subspace expansion available on git (there are four of them).
None of them work correctly, in the following sense:
If I start from a  1 x 2 x 1 tensor initially, then AL is updated to a 2 x 2 x 1 while AR is updated to a 1 x 2 x 2.
For both side, we cannot chain the AL or the AR. Any attempt to further expand the subspaces fail.

**Comment by mtfishman** [comment]
Oh I see, thanks for investigating. I guess we have to make a special case for the 1-site unit cell and expand both link spaces of AL and AR at once with the same expansion.

**Comment by LHerviou** [comment]
Updating a PR

**Comment by mtfishman** [comment]
Closed by #44.
